test/all.test.js

```diff
@@ -51,6 +51,69 @@ describe(Support.getTestDialectTeaser('Tests'), function () {
 });
 
 function tests() {
+	describe('underscored:true', function() {
+		/*jshint camelcase: false */
+		it('underscores fieldNames & foreignKeys', function() {
+			this.folder = this.sequelize.define('folder', {
+				name: Sequelize.STRING
+			}, {
+				underscored: true
+			});
+			this.folder.isHierarchy();
+			expect(this.folder.attributes).to.have.property('hierarchy_level');
+			expect(this.folder.attributes).to.have.property('parent_id');
+			expect(this.folder.associations.ancestors.through.model.attributes).to.have.property('folder_id');
+			expect(this.folder.associations.ancestors.through.model.attributes).to.have.property('ancestor_id');
+			expect(this.folder.associations.ancestors.foreignKey).to.equal('folder_id');
+			expect(this.folder.associations.children.foreignKey).to.equal('parent_id');
+		});
+		it('doesn\'t override user supplied names', function() {
+			this.folder = this.sequelize.define('folder', {
+				name: Sequelize.STRING
+			}, {
+				underscored: true
+			});
+			this.folder.isHierarchy({
+				levelFieldName: 'testFieldName',
+				foreignKey: 'testForeignKey',
+				throughKey: 'testThroughKey',
+				throughForeignKey: 'testThroughForeignKey'
+			});
+
+			expect(this.folder.attributes).to.have.property('testFieldName');
+			expect(this.folder.attributes).to.have.property('testForeignKey');
+			expect(this.folder.associations.ancestors.through.model.attributes).to.have.property('testThroughForeignKey');
```

Comment:
No, I copied
```
expect(this.folder.associations.ancestors.through.model.attributes).to.have.property('folder_id');
expect(this.folder.associations.ancestors.through.model.attributes).to.have.property('ancestor_id');
```
And changed those property names to `testThroughForeignKey` for the "doesn't override user supplied names" test. Changing those property names made made one of the assertions redundant. I'll remove one of them.